### PR TITLE
bakery: add macaroon_to_dict convenience function

### PR DESCRIPTION
**Pull request by rogpeppe**
This makes it easier serialize macaroons in a list (for example).

### CODE REVIEW

File: macaroonbakery/tests/test_utils.py [hunk 1]
```diff
@@ -19,3 +23,14 @@ def test_cookie_expires_with_timezone(self):
         self.assertRaises(
             ValueError, cookie, 'http://example.com', 'test', 'value',
             expires=timestamp)
+
+
+class MacaroonToDictTest(TestCase):
+    def test_macaroon_to_dict(self):
+        m = pymacaroons.Macaroon(
+            key=b'rootkey', identifier=b'some id', location='here', version=2)
+        as_dict = bakery.macaroon_to_dict(m)
```
Comment (mitechie) [inline macaroonbakery/tests/test_utils.py]: it's preferred to just implement the dict interface and to then just cast it to a dict vs a named function for this. Is that an option? If I recall you'd just implement __iter__ on the object.

File: macaroonbakery/tests/test_utils.py [hunk 2]
```diff
@@ -19,3 +23,14 @@ def test_cookie_expires_with_timezone(self):
...
+        data = json.dumps(as_dict)
```
Comment (mitechie) [inline macaroonbakery/tests/test_utils.py]: then this is just json.dumps(bakery) vs the result of the dict call.

**Comment by mitechie** [comment]
https://docs.python.org/3/reference/datamodel.html?emulating-container-types#emulating-container-types for the whole shebang of acting like a dict. To be serializ-able I still think you just need __iter__ but dbl check.

File: macaroonbakery/tests/test_utils.py (hunk 1 above)
Comment (rogpeppe) [reply macaroonbakery/tests/test_utils.py]: Unfortunately, as discussed online, this isn't an option, as you can't just make arbitrary objects implement custom JSON encoding without changing the caller too.

**Pull request merged**